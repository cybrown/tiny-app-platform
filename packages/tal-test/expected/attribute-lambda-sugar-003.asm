main()
  entry:
    Literal           0
    MakeArray
    Literal           0
    MakeRecord
    Local             get_record
    Call
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Literal           "parent"
    Literal           1
    FunctionRef       name: func_0
    Index
    Index
    Attribute         name: name
    Call
func_0($$arg$$)
  entry:
    Local             $$arg$$
    Attribute         name: people
