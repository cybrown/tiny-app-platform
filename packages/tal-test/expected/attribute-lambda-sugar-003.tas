get_record() | .people[1]["parent"].name
