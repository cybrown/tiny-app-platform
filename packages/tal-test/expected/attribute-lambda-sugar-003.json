[
  {
    "kind": "Pipe",
    "first": {
      "kind": "Call",
      "args": [],
      "value": {
        "kind": "Local",
        "name": "get_record"
      }
    },
    "values": [
      {
        "kind": "Attribute",
        "key": "name",
        "value": {
          "kind": "Index",
          "index": "parent",
          "value": {
            "kind": "Index",
            "index": 1,
            "value": {
              "kind": "AttributeLambdaSugar",
              "key": "people"
            }
          }
        }
      }
    ]
  }
]